src/main/java/seedu/address/model/self/Height.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.self;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the user's height in the diet tracker of EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidHeight(String)}
+ */
+public class Height {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Height should only contain numbers, and it should not be blank. It is to be input in centimeters.";
+
+    /*
+     * The first character of the height must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
+
+    public final String height;
+    public final Float ht;
+
+    /**
+     * Constructs a {@code Height}.
+     *
+     * @param height A valid height.
+     */
+    public Height(String height) {
+        requireNonNull(height);
+        checkArgument(isValidHeight(height), MESSAGE_CONSTRAINTS);
+        this.height = height.toString();
```

Comment:
No need to call toString on the height as it is already a String. Causing Codacy errors.